fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,259 @@
+    "The list of metrics and list of conf_intervals are not the same length."
+
+_METRICS_NOT_LIST_OR_STR_ERROR = \
+    """Metric should be a string of a single metric or a list of the metrics
+     in provided MetricFrame, but {0} was provided"""
+_ERROR_BARS_MUST_BE_TUPLE = "Calculated error_bars must be tuple of length 2"
+_CONF_INTERVALS_MUST_BE_TUPLE = "Calculated conf_intervals must be tuple of length 2"
+_ERROR_BARS_NEGATIVE_VALUE_ERROR = "Calculated relative errors from error_bars must be positive"
+_CONF_INTERVALS_FLIPPED_BOUNDS_ERROR = \
+    "Calculated conf_intervals' upper bound cannot be less than lower bound"
+
+
+def _check_if_metrics_and_error_metrics_same_length(metrics, error_bars, conf_intervals):
+    if error_bars is not None:
+        if len(error_bars) != len(metrics):
+            raise ValueError(_METRIC_AND_ERROR_BARS_NOT_SAME_LENGTH_ERROR)
+    elif conf_intervals is not None:
+        if len(conf_intervals) != len(metrics):
+            raise ValueError(_METRIC_AND_CONF_INTERVALS_NOT_SAME_LENGTH_ERROR)
+
+
+def _check_for_ambiguous_error_metric(error_bars, conf_intervals):
+    # ambiguous error metric if both are provided
+    if error_bars is not None and conf_intervals is not None:
+        raise ValueError(_GIVEN_BOTH_ERROR_BARS_AND_CONF_INT_ERROR)
+
+
+def _check_for_valid_metrics_format(metrics):
+    # ensure metrics is either list, str, or None
+    if not (isinstance(metrics, list) or isinstance(metrics, str) or metrics is None):
+        raise ValueError(_METRICS_NOT_LIST_OR_STR_ERROR.format(type(metrics)))
+
+
+def _check_valid_error_bars(df_error_bars):
+    for tup in df_error_bars:
+        if not isinstance(tup, tuple) or len(tup) != 2:
+            raise ValueError(_ERROR_BARS_MUST_BE_TUPLE)
+        if not all(ele >= 0 for ele in tup):
+            raise ValueError(_ERROR_BARS_NEGATIVE_VALUE_ERROR)
+
+
+def _check_valid_conf_interval(df_conf_intervals):
+    for tup in df_conf_intervals:
+        if not isinstance(tup, tuple) or len(tup) != 2:
+            raise ValueError(_CONF_INTERVALS_MUST_BE_TUPLE)
+        if tup[0] > tup[1]:
+            raise ValueError(_CONF_INTERVALS_FLIPPED_BOUNDS_ERROR)
+
+
+def plot_metric_frame(metric_frame: MetricFrame,
+                      kind: str = "scatter",
+                      metrics: Union[List[str], str] = None,
+                      error_bars: Union[List[str], str] = None,
+                      conf_intervals: Union[List[str], str] = None,
+                      axs=None,
+                      show_plot=True,
+                      plot_error_labels: bool = False,
+                      text_precision_digits: int = 4,
+                      text_fontsize: int = 8,
+                      text_color: str = "black",
+                      text_ha: str = "center",
+                      capsize=10,
+                      colormap="Pastel1",
+                      subplot_shape=None,
+                      figsize=None
+                      ):
```

Comment:
Shouldn't this have a `-> matplotlib.axes.Axes` or something like that to show the return value?